aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -80,12 +82,18 @@ struct WelfordOps {
       new_count
     };
   }
-  inline C10_DEVICE scalar_t project(acc_t acc) const {
+  inline C10_DEVICE res_t project(acc_t acc) const {
+    auto mean = acc.mean;
     combine_t divisor = unbiased ? (acc.nf - 1) : acc.nf;
     auto ret = (divisor > 0) ?
       (take_sqrt ? device_sqrt(acc.m2 / divisor) : (acc.m2 / divisor))
       : NAN;
-    return (scalar_t) ret;
+#if defined(__CUDACC__) || defined(__HIPCC__)
```

Comment:
Why are we using a tuple for CUDA, and a vector for CPU? Can't we use a tuple for both? (`std::tuple` on CPU and `thrust::tuple` on CUDA)

Reply:
Ideally it should be vector for both CPU and CUDA, but CUDA doesn't support to call thrust::vector methods on DEVICE. So I use tuple instead.

Reply:
I don't like using `std::vector` here for three reasons:

(1) It is worse for performance, because it requires an allocation on the heap (this is not a huge deal since we are not calling `project` in any tight loop, but it's still better to avoid if possible.)

(2) It is confusing and error-prone to have different behavior on CUDA vs. CPU.

(3) `std::vector` does not allow multiple different types (someday we might have an op that returns a tuple of (FloatTensor, LongTensor) for example)

Reply:
Fixed. Made a tuple on both CPU and CUDA parts